esrally/telemetry.py

```diff
@@ -1774,8 +1775,10 @@ def on_benchmark_start(self):
             self.logger.exception("Could not retrieve cluster version info")
             return
         distribution_flavor = client_info["version"].get("build_flavor", "oss")
-        # build hash will only be available on serverless if the client has operator privs
+        # serverless returns dummy build hash which gets overridden when running with operator privileges
```

Comment:
I'd add a `# TODO` here that we'll likely refactor `Telemetry()` to accept the config object and allow handling the `revision` based on `driver.serverless` settings

Reply:
Addressed in https://github.com/elastic/rally/pull/1756/commits/bbd8cabe478798b71b6623526f4a22f356af8eb5.